$NetBSD: patch-aa,v 1.4 2013/04/28 03:15:37 dholland Exp $

- configure for pkgsrc
- use libtool (why?)
- rearrange installation
- honor PKGMANDIR

--- src/Makefile.orig	2008-06-27 03:33:36.000000000 +0000
+++ src/Makefile
@@ -7,20 +7,19 @@
 
 ############## Public section. Modify according to your needs.  ###############
 
-CC      = gcc
-RM      = rm -f
-CP      = cp
+#CC      = gcc
+#RM      = rm -f
+#CP      = cp
 
-DESTDIR = /usr/local
 IPATH   = 
 LPATH   = 
-OPTI    = -Wall -O3 -funroll-loops -fomit-frame-pointer #-DDEBUG_FCN
+OPTI    = -Wall #-O3 -funroll-loops -fomit-frame-pointer #-DDEBUG_FCN
 DBUG    = #-ggdb -pg -fbounds-check
 WARN	= #-W -Wall #-pedantic -ansi
 
 ### Unix version ###
-CFLAGS  = ${OPTI} ${DBUG} ${WARN} ${IPATH} `pkg-config gtk+-2.0 --cflags`
-LDFLAGS = ${OPTI} ${DBUG} ${WARN} ${LPATH} `pkg-config gtk+-2.0 --libs`
+CFLAGS  += `pkg-config gtk+-2.0 --cflags` -prefer-pic
+LDFLAGS += `pkg-config gtk+-2.0 --libs`
 NAME    = beaver
 
 ### Windows version ###
@@ -32,66 +31,38 @@ NAME    = beaver
 
 ###################### Private section. Do not modify  ########################
 
-ifndef DBUG
-	STRIP = strip --strip-all ${NAME}
-endif
-
 SRC	= main.c search.c tools.c languages.c editor.c prefs.c conf.c msgbar.c\
 	  toolbar.c interface.c filesops.c completion.c undoredo.c
 INC	= ${SRC:.c=.h} struct.h wordfile.h
 OBJ	= ${SRC:.c=.o}
 
 all	: ${OBJ}
-	${CC} -o ${NAME} ${OBJ} ${LDFLAGS}
-	@${STRIP}
+	${LIBTOOL} --mode=link --tag=CC \
+		${CC} -o ${NAME} ${OBJ:.o=.lo} ${LDFLAGS}
 
 .c.o	:
-	${CC} ${CFLAGS} -c $< -o $@
+	${LIBTOOL} --mode=compile --tag=CC \
+		${CC} ${CFLAGS} -c $< -o ${@:.o=.lo}
 
 ${OBJ}	: ${INC}
 
 install :
-	install -d $(DESTDIR)/share/beaver/bin
-	install beaver $(DESTDIR)/share/beaver/bin
-	$(CP) $(DESTDIR)/share/beaver/bin/beaver $(DESTDIR)/share/beaver/bin/beaver.orig
-	echo "#!/bin/bash" > $(DESTDIR)/share/beaver/bin/beaver
-	echo "cd $(DESTDIR)/share/beaver/bin/" >> $(DESTDIR)/share/beaver/bin/beaver
-	echo "./beaver.orig" >> $(DESTDIR)/share/beaver/bin/beaver
-	ln -sf $(DESTDIR)/share/beaver/bin/beaver $(DESTDIR)/bin/beaver
-	install -d $(DESTDIR)/share/beaver/bl
-	$(CP) ../bl/*.bl $(DESTDIR)/share/beaver/bl
-	install -d $(DESTDIR)/share/beaver/pixmaps
-	$(CP) ../pixmaps/beaver.png $(DESTDIR)/share/beaver/pixmaps
-	$(CP) ../pixmaps/about.xpm $(DESTDIR)/share/beaver/pixmaps
-	chmod 644 $(DESTDIR)/share/beaver/pixmaps/beaver.png
-	chmod 644 $(DESTDIR)/share/beaver/pixmaps/about.xpm
-	install -d $(DESTDIR)/share/applications
-	$(CP) ../beaver.desktop $(DESTDIR)/share/applications/beaver.desktop
-	chmod 644 $(DESTDIR)/share/applications/beaver.desktop
-	sed -i $(DESTDIR)/share/applications/beaver.desktop -e "s/Icon=/Icon=`echo $(DESTDIR) | sed -e 's/\//\\\\\//g'`\/share\/beaver\/pixmaps\//g" 
-	gzip -c9 ../beaver.1x > ../beaver.1x.gz
-	install -d $(DESTDIR)/man/man1
-	$(CP) ../beaver.1x.gz $(DESTDIR)/man/man1
-	chmod 644 $(DESTDIR)/man/man1/beaver.1x.gz
-	@echo
-	@echo "Files installed :"
-	@echo "-----------------"
-	@echo
-	@echo "$(DESTDIR)/bin/beaver"
-	@echo "$(DESTDIR)/bin/beaver.orig"
-	@echo "$(DESTDIR)/share/beaver/bin/beaver"
-	@echo "$(DESTDIR)/share/beaver/bl/example1.bl"
-	@echo "$(DESTDIR)/share/beaver/bl/example2.bl"
-	@echo "$(DESTDIR)/share/beaver/bl/glib.bl"
-	@echo "$(DESTDIR)/share/beaver/pixmaps/beaver.png"
-	@echo "$(DESTDIR)/share/beaver/pixmaps/about.xpm"
-	@echo "$(DESTDIR)/share/applications/beaver.desktop"
-	@echo "$(DESTDIR)/man/man1/beaver.1x.gz"
-	@echo
-	@echo ",----------------------------------."
-	@echo "| Beaver succesfully installed ^_^ |"
-	@echo "\`----------------------------------'"
-
+	${BSD_INSTALL_DATA_DIR} $(DESTDIR)${PREFIX}/bin
+	${LIBTOOL} --mode=install ${BSD_INSTALL_PROGRAM} \
+		beaver $(DESTDIR)${PREFIX}/bin/beaver
+	${BSD_INSTALL_DATA_DIR} $(DESTDIR)${PREFIX}/share/beaver/bl
+	${BSD_INSTALL_DATA} ../bl/*.bl $(DESTDIR)${PREFIX}/share/beaver/bl
+	${BSD_INSTALL_DATA_DIR} $(DESTDIR)${PREFIX}/share/pixmaps/beaver
+	${BSD_INSTALL_DATA} ../pixmaps/beaver.png \
+		$(DESTDIR)${PREFIX}/share/pixmaps/beaver
+	${BSD_INSTALL_DATA} ../pixmaps/about.xpm \
+		$(DESTDIR)${PREFIX}/share/pixmaps/beaver
+	${BSD_INSTALL_DATA_DIR} $(DESTDIR)${PREFIX}/share/applications
+	${BSD_INSTALL_DATA} ../beaver.desktop \
+		$(DESTDIR)${PREFIX}/share/applications/beaver.desktop
+	${BSD_INSTALL_MAN_DIR} $(DESTDIR)${PREFIX}/${PKGMANDIR}/man1
+	${BSD_INSTALL_MAN} ../beaver.1x \
+		$(DESTDIR)${PREFIX}/${PKGMANDIR}/man1/beaver.1
 .PHONY: clean re
 
 clean	:
